plothist/plotters.py

```diff
@@ -758,6 +781,61 @@ def plot_comparison(
     return ax
 
 
+def savefig(fig, path, new_figsize=None, adjust_after_tight_layout=False):
+    """
+    Save a Matplotlib figure with consistent figsize, axes size and subplot spacing.
+
+    Parameters
+    ----------
+    fig : matplotlib.figure.Figure
+        The Matplotlib figure to be saved.
+    path : str
+        The output file path where the figure will be saved.
+    new_figsize : tuple, optional
+        The new figsize as a (width, height) tuple. If None, the original figsize is preserved.
+    adjust_after_tight_layout : bool, optional
+        Adjust the space between the subplots just before saving. Useful for 2D plots.
+
+    Returns
+    -------
+    None
+    """
+    old_width, old_height = fig.get_size_inches()
+
+    if new_figsize is not None:
+        width_scale = new_figsize[0] / old_width
+        height_scale = new_figsize[1] / old_height
+    else:
+        width_scale = 1.0
+        height_scale = 1.0
+
+    axes = fig.get_axes()
+    axes_dimensions = [
+        (pos.width / width_scale, pos.height / height_scale)
+        for pos in [ax.get_position() for ax in axes]
+    ]
+    wspace, hspace = fig.subplotpars.wspace, fig.subplotpars.hspace
+
+    fig.tight_layout()
+    fig.subplots_adjust(wspace=wspace, hspace=hspace)
+
+    for k_ax, ax in enumerate(axes):
+        ax.set_position(
+            Bbox.from_bounds(
+                ax.get_position().x0,
+                ax.get_position().y0,
+                axes_dimensions[k_ax][0],
+                axes_dimensions[k_ax][1],
+            )
+        )
+    if adjust_after_tight_layout:
```

Comment:
why not always doing it?

Reply:
Always doing it can mess up some plots. Not doing it cannot mess up plots, only make some subplots misaligned on a very few examples.